src/services/db.js

```diff
@@ -82,40 +98,42 @@ export default class Database {
     return this.db.userTricks.clear();
   };
 
-  // populate the databes with tricks from the predefinedTricks.js
-  populateTricks = () => {
-    return this.db.predefinedTricks.clear().then(() => {
-      console.log("updating predefinedTricks")
-      const trickList = Papa.parse(predefinedTricks, {dynamicTyping: true}).data;
-
-      // this uses the labels of the csv but, also adds an id
-      const header = trickList.shift().concat(["stickFrequency"]);
-
-      const tricks = trickList.map(trick => {
-        // add 0 for stickFrequency
-        trick.push(0);
-        // make key value pairs
-        return Object.assign.apply({},
-          header.map((v,i) => {
+  // populate the database with tricks from the predefinedTricks.js
+  populateTricks = async () => {
+    await this.db.predefinedTricks.clear()
+
+    const trickList = Papa.parse(predefinedTricks, {dynamicTyping: true}).data;
+
+    // this uses the labels of the csv but, also adds an id
+    const header = trickList.shift().concat(["stickFrequency"]);
+
+    const tricks = trickList.map(trick => {
+      // add 0 for stickFrequency
+      trick.push(0);
+      // make key value pairs
+      return Object.assign.apply({},
+          header.map((v, i) => {
             return ({[v]: trick[i]})
           })
-      );});
-
-      //this turns the list of recommendedPrerequisites (which are separated by an ;) into an Array
-      tricks.map(trick => {
-        if (typeof trick.recommendedPrerequisites === "string") {
-          trick.recommendedPrerequisites = trick.recommendedPrerequisites.split(";").map(string => Number(string));
-        } else if (typeof trick.recommendedPrerequisites === "number") {
-          trick.recommendedPrerequisites = [trick.recommendedPrerequisites];
-        }
-      });
+      );
+    });
 
-      // adds the tricks to the database
-      return this.db.predefinedTricks.bulkPut(tricks).then(() => {
-        console.log("added tricks to database from the csv");
-        this.db.versions.put({"key": "predefinedTricksVersion", "version": predefinedTricksVersion});
-      });
+    // This turns the list of recommendedPrerequisites (which are separated by a ";") into an Array
+    tricks.map(trick => {
+      if (typeof trick.recommendedPrerequisites === "string") {
+        trick.recommendedPrerequisites = trick.recommendedPrerequisites.split(";").map(string => Number(string));
```

Comment:
This is a change from the previous code, so feel free to disregard, but I just wanted to throw this into here.

architecture: Does this need to be a semicolon-separated String? Does the underlying DB support Arrays?

Also: `trick.recommendedPrerequisites` of type `string` gets overwritten with `number[]`
Types without TypeScript are already hard – this makes it needlessly complicated. 
Depending on the time of access, this property is either a `string`, a `number` or a `number[]`.
Could an intermediary object be used here?

Reply:
I am managing all the tricks in a google sheet and using a semicolon-separated string seemed like an easy option. However, if there is only one recommendedPrerequisite this is read as a number, which leads to the code I implemented here.
I agree that the types here are quite messy.
Do you see a better option of implementing this?

Reply:
I think the best solution would be to set `trick.recommendedPrerequisites` in one location only and use other intermediary variables for conversion. This way, there is only one place where change happens. Also makes it easier to maintain later on. 

On the point of maintainability I would also strongly suggest to make use of TypeScript.
I helped @JulianDietzel here and there with some code suggestions on the repo, and we found that trying to infer types from usage caused a significant slowdown during writing code — especially in cases where a variable can be of multiple types.

If you are interested, I could look into moving the repo to TypeScript and Vite.

Reply:
> I am managing all the tricks in a google sheet and using a semicolon-separated string seemed like an easy option

Rollup (and as a result, Vite) has the [dvs loader](https://github.com/rollup/plugins/tree/master/packages/dsv#readme) Plugin which allows you to simply import a CSV File as a JS Module. I would assume Webpack has something similar. 
This only works for compile-time data. If you are fetching that data from the browser from Google Sheet's API, this won't be useful.

Reply:
Talking to @WaldemarLehner we came to the following conclusion:
- Parsing the CSV File to a JS Object at compile time like he recommended would be nice (as then the conversion does not have to happen every time someone enters the website for the first time) but quite out of scope her
- The way it is handled at the moment is actually alright, considering how the data is passed from the Google Doc via a CSV into the db and it does not necessarily need to be changed
- The code of the data cleaning could be written a good bit cleaner. So I did that and pushed the changes as a new commit ([710df3a](https://github.com/bastislack/highline-freestyle/pull/233/commits/710df3a3746e36b7f9297e0f43110b52cccfef97))

Reply:
> I think the best solution would be to set trick.recommendedPrerequisites in one location only and use other intermediary variables for conversion. This way, there is only one place where change happens. Also makes it easier to maintain later on.

That totally makes sense.

> On the point of maintainability I would also strongly suggest to make use of TypeScript.
I helped @JulianDietzel here and there with some code suggestions on the repo, and we found that trying to infer types from usage caused a significant slowdown during writing code — especially in cases where a variable can be of multiple types.

You are right, that could definitely save some headache with runtime errors and development in general. I will open an issue so we can discuss the migration to TypeScript.